src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/KpiService.kt

```diff
@@ -18,31 +18,33 @@ class KpiService(
   private val offenderSearchService: OffenderSearchService,
 ) {
 
-  fun getNumberOfReviewsConductedAndPrisonersReviewed(day: LocalDate): ReviewsConductedPrisonersReviewed = runBlocking {
-    kpiRepository.getNumberOfReviewsConductedAndPrisonersReviewed(day)
+  suspend fun getNumberOfReviewsConductedAndPrisonersReviewed(day: LocalDate): ReviewsConductedPrisonersReviewed {
+    return kpiRepository.getNumberOfReviewsConductedAndPrisonersReviewed(day)
   }
 
-  fun getNumberOfPrisonersOverdue(): Int = runBlocking {
+  suspend fun getNumberOfPrisonersOverdue(): Int {
     // Get a set containing all the prisonerNumber of everyone in prison (any prison)
     val allPrisoners = mutableSetOf<String>()
-    val prisonersInPrisons = getPrisons().map { prisonId -> async { getPrisonersInPrison(prisonId) } }.awaitAll()
-    prisonersInPrisons.forEach { prisoners ->
+    getPrisons().map { prisonId ->
+      getPrisonersInPrison(prisonId)
+    }.collect { prisoners ->
       allPrisoners.addAll(prisoners.map(OffenderSearchPrisoner::prisonerNumber))
     }
 
     println("Number of prisoners across the estate: ${allPrisoners.size}")
 
     // Get prisoner numbers overdue, "filter" out people no longer in prison and return the count
-    kpiRepository
+    return kpiRepository
       .getPrisonerNumbersOverdueReview()
       .count { allPrisoners.contains(it) }
   }
 
-  private fun getPrisons(): List<String> = runBlocking {
+  private suspend fun getPrisons(): Flow<String> {
```

Comment:
`getPrisons` was called from inside a coroutine so should avoid using `runBlocking`